x/msg_delegation/spec/msg_delegation.md

```diff
@@ -0,0 +1,125 @@
+# `msg_delegation` Module
+
+The `msg_delegation` module provides support for granting arbitrary capabilities
+from one account (the granter) to another account (the grantee). Capabilities
+must be granted for a particular type of `sdk.Msg` one by one using an implementation
+of `Capability`.
+
+## Types
+
+Capabilities determine exactly what action is delegated. They are extensible
+and can be defined for any sdk.Msg type even outside of the module where the Msg is defined.
+
+### Capability
+
+```go
+type Capability interface {
+	// MsgType returns the type of Msg's that this capability can accept
```

Comment:
I was wondering if `Expiration` is so generic that it could be part of this interface as well?
When moved into the concrete Capabilty implementation the where to store it, is solved

Reply:
Well I thought it was so generic to not make it part of the Capability interface - i.e. that you might want to always allow an expiration regardless of whether capabilities explicitly support it. So the simplest way to handle this seemed to make it top-level for every capability.